### PR TITLE
Even shorter notation to build and run a Webhosts

### PR DESCRIPTION
**Pull request by faniereynders**
Added support for allowing a shorthand notation to build and run a Webhost with the default configuration including a typed Startup class

### CODE REVIEW

**Comment by faniereynders** [comment]
@davidfowl?

**Comment by muratg** [comment]
@JunTaoLuo @davidfowl could you guys take a look?

File: src/Microsoft.AspNetCore/WebHost.cs [hunk 1]
```diff
@@ -191,5 +191,40 @@ public static IWebHostBuilder CreateDefaultBuilder(string[] args)
 
             return builder;
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebHostBuilder"/> class with pre-configured defaults using typed Startup
+        /// </summary>
+        /// <typeparam name="T">Specify the startup type to be used by the web host.</typeparam>
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore/WebHost.cs]: I would prefer using the same language as `UseStartup` https://github.com/aspnet/Hosting/blob/70a3017b12b4e9648f65acea2c9ad37e383763b0/src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs#L77

File: src/Microsoft.AspNetCore/WebHost.cs [hunk 2]
```diff
@@ -191,5 +191,40 @@ public static IWebHostBuilder CreateDefaultBuilder(string[] args)
...
+        /// <param name="args">The command line args.</param>
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore/WebHost.cs]: `summary` first, then `remarks`, then `params` and lastly `returns`.

File: src/Microsoft.AspNetCore/WebHost.cs [hunk 3]
```diff
@@ -191,5 +191,40 @@ public static IWebHostBuilder CreateDefaultBuilder(string[] args)
...
+        /// <remarks>
+        ///   The following defaults are applied to the returned <see cref="WebHostBuilder"/>:
+        ///     use Kestrel as the web server,
+        ///     set the <see cref="IHostingEnvironment.ContentRootPath"/> to the result of <see cref="Directory.GetCurrentDirectory()"/>,
+        ///     load <see cref="IConfiguration"/> from 'appsettings.json' and 'appsettings.[<see cref="IHostingEnvironment.EnvironmentName"/>].json',
+        ///     load <see cref="IConfiguration"/> from User Secrets when <see cref="IHostingEnvironment.EnvironmentName"/> is 'Development' using the entry assembly,
+        ///     load <see cref="IConfiguration"/> from environment variables,
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore/WebHost.cs]: We also load configuration from supplied command line args.

File: src/Microsoft.AspNetCore/WebHost.cs [hunk 4]
```diff
@@ -191,5 +191,40 @@ public static IWebHostBuilder CreateDefaultBuilder(string[] args)
...
+            return WebHost.CreateDefaultBuilder(args)
+                .UseStartup<T>();
+        }
+
+        /// <summary>
+        /// Builds an Microsoft.AspNetCore.Hosting.IWebHost which hosts a web application and
+        /// runs a web application and block the calling thread until host shutdown.
+        /// </summary>
+        /// <param name="builder">The <see cref="IWebHostBuilder"/></param>
+        public static void BuildAndRun(this IWebHostBuilder builder)
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore/WebHost.cs]: I'm not sure if we want to add this extension. It doesn't save that much compared to the original `Build().Run()`. Though if we were to add this extension, we would also need to add `BuildAndRunAsync()`.

File: test/TestSites/CreateDefaultBuilderApp/Startup.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.AspNetCore.Builder;
```
Comment (JunTaoLuo) [inline test/TestSites/CreateDefaultBuilderApp/Startup.cs]: Add copyright header

File: test/TestSites/CreateDefaultBuilderApp/Program.cs [hunk 6]
```diff
@@ -36,6 +36,11 @@ static void Main(string[] args)
                     });
                 })
                 .Build().Run();
+
+            Console.ReadKey();
```
Comment (JunTaoLuo) [inline test/TestSites/CreateDefaultBuilderApp/Program.cs]: This doesn't add any coverage for the new overload other than for compilation. Create a copy of this app that uses `CreateDefaultBuilder<Startup>` and add an test like https://github.com/aspnet/MetaPackages/blob/dev/test/Microsoft.AspNetCore.FunctionalTests/WebHostFunctionalTests.cs#L51 instead.

**Comment by JunTaoLuo** [comment]
Thanks @faniereynders for the contribution! Your changes were merged in https://github.com/aspnet/MetaPackages/commit/07bf8d02a054e37461c8ccf50dc95ee594556c62.

**Pull request closed without merging**